contracts/test/behaviour/sfcStakingStrategy.js

```diff
@@ -0,0 +1,527 @@
+        4,
+        "Incorrect Supported validators length"
+      );
+
+      for (const validatorId of testValidatorIds) {
+        expect(
+          await sonicStakingStrategy.isSupportedValidator(validatorId)
+        ).to.equal(true, "Validator expected to be supported");
+      }
+
+      expect(await sonicStakingStrategy.platformAddress()).to.equal(
+        addresses.SFC,
+        "Incorrect platform address set"
+      );
+
+      expect(await sonicStakingStrategy.vaultAddress()).to.equal(
+        oSonicVault.address,
+        "Incorrect Vault address"
+      );
+
+      expect(await sonicStakingStrategy.harvesterAddress()).to.equal(
+        AddressZero,
+        "Harvester address not empty"
+      );
+
+      expect(
+        (await sonicStakingStrategy.getRewardTokenAddresses()).length
+      ).to.equal(0, "Incorrectly configured Reward Token Addresses");
+
+      expect(await oSonicVault.priceProvider()).to.not.equal(
+        AddressZero,
+        "Price provider address not set"
+      );
+
+      expect(await oSonicVault.priceUnitMint(wS.address)).to.equal(
+        oethUnits("1"),
+        "not expected PriceUnitMint"
+      );
+    });
+  });
+
+  describe("Deposit/Delegation", function () {
+    it("Should fail when unsupported functions are called", async () => {
+      const { sonicStakingStrategy, timelock, wS } = await context();
+
+      await expect(
+        sonicStakingStrategy
+          .connect(timelock)
+          .setPTokenAddress(wS.address, wS.address)
+      ).to.be.revertedWith("unsupported function");
+
+      await expect(
+        sonicStakingStrategy.connect(timelock).collectRewardTokens()
+      ).to.be.revertedWith("unsupported function");
+
+      await expect(
+        sonicStakingStrategy.connect(timelock).removePToken(wS.address)
+      ).to.be.revertedWith("unsupported function");
+    });
+
+    it("Should accept and handle S token allocation and delegation to SFC", async () => {
+      const amount = oethUnits("15000");
+      await depositTokenAmount(amount);
+    });
+
+    it("Should earn rewards as epochs pass", async () => {
+      const { sonicStakingStrategy, wS } = await context();
+
+      const amount = oethUnits("15000");
+      await depositTokenAmount(amount);
+
+      const balanceBefore = await sonicStakingStrategy.checkBalance(wS.address);
+      await advanceSfcEpoch(1);
+      expect(await sonicStakingStrategy.checkBalance(wS.address)).to.be.gt(
+        balanceBefore
+      );
+    });
+
+    it("Should be able to restake the earned rewards", async () => {
+      const { sonicStakingStrategy, wS, testValidatorIds } = await context();
+
+      const amount = oethUnits("15000");
+      await depositTokenAmount(amount);
+      await advanceSfcEpoch(1);
+
+      const stratBalanceBefore = await sonicStakingStrategy.checkBalance(
+        wS.address
+      );
+
+      await sonicStakingStrategy.restakeRewards(testValidatorIds);
```

Comment:
❓This is a very important test - it's critical the balance does not change. Are we sure that there are actually rewards that are being restaked here?

Reply:
good point: https://github.com/OriginProtocol/origin-dollar/pull/2335/commits/4ea53b1dcec276fb1aca2f3c5db7ce9b52e72994